src/solver/batchnorm/forward_inference_ck.cpp

```diff
@@ -126,52 +145,47 @@ template <typename XDataType,
           typename ScaleDataType,
           typename BiasDataType,
           typename MeanVarDataType>
-static void RunCKSolution(const Handle& handle,
-                          const AnyInvokeParams& primitive_parameters,
-                          const miopen::batchnorm::ProblemDescription& problem)
+ConvSolution InvokerFactoryMakerNHWC(const miopen::batchnorm::ProblemDescription& bn_problem)
 {
-    const auto& args = CKArgsBNormFwd{problem};
-
-    using DeviceOp = ck::tensor_operation::device::DeviceElementwise<
-        ck::Tuple<XDataType, MeanVarDataType, MeanVarDataType, ScaleDataType, BiasDataType>,
-        ck::Tuple<YDataType>,
-        Normalize,
-        Rank>;
-    const auto bn_fwd_ptrs = ck::tensor_operation::device::instance::DeviceOperationInstanceFactory<
-        DeviceOp>::GetInstances();
+    ConvSolution result;
+    result.invoker_factory = [args         = CKArgsBNormFwd{bn_problem},
+                              kernel_index = CheckCKApplicability<XDataType,
+                                                                  YDataType,
+                                                                  AccDataType,
+                                                                  ScaleDataType,
+                                                                  BiasDataType,
+                                                                  MeanVarDataType>(bn_problem)](
+                                 const std::vector<Kernel>& /*kernels*/) mutable {
+        return [args = std::move(args), kernel_index = kernel_index](
+                   const Handle& handle, const AnyInvokeParams& primitive_parameters) {
+            using DeviceOp = ck::tensor_operation::device::DeviceElementwise<
```

Comment:
Sorry, I missed this earlier. Please move lines (161-171), i.e. computation of `bn_ptr` outside and capture `bn_ptr` by `move` semantics. See the `InitInvokerFactory` for convolution solvers.

Reply:
@amberhassaan , changes have been checked in. Thanks